packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -0,0 +1,157 @@
+	Open = 0xFF
+}
+
+/**
+ * @publicAPI
+ */
+export enum BarrierPosition {
+	Closed = 0x00,
+	Closing = 0xFC,
+	Stopped = 0xFD
+	Openning = 0xFE,
+	Open = 0xFF,
+}
+
+// @publicAPI
+export enum SignalType {
+	"Not Supported" = 0x00,
+	Audible = 0x01,
+	Visual = 0x02,
+}
+
+@commandClass(CommandClasses.BarrierOperator)
+@implementedVersion(1)
+export class BarrierOperatorCC extends CommandClass {
+	declare ccCommand: BarrierOperatorCommand;
+}
+
+// To be reviewed as CC should only accept 0x00 or 0xFF, if more work needed
+@CCCommand(BarrierOperatorCommand.Set)
+export class BarrierOperatorCCSet extends BarrierOperatorCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | BarrierOperatorCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			validatePayload(this.payload.length >= 1);
+			this.targetValue = this.payload[0];
+		} else {
+			this.targetValue = options.targetValue;
+		}
+	}
+
+	public targetValue: number;
+
+	public serialize(): Buffer {
+		this.payload = Buffer.from([this.targetValue]);
+		return super.serialize();
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: { "target value": this.targetValue },
+		};
+	}
+}
+
+/** Returns the State  */
+interface BarrierOperatorCCGetOptions extends CCCommandOptions {
+	someProperty: number;
+}
+
+@CCCommand(BarrierOperatorCommand.Get)
+@expectedCCResponse(BarrierOperatorCCReport)
+export class BarrierOperatorCCGet extends BarrierOperatorCC {}
+
+@CCCommand(BarrierOperatorCommand.Report)
+export class BarrierOperatorCCReport extends BarrierOperatorCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+		
+		// return raw value if cannot decode
+		this._barrierPosition = this.payload[0];
+		this.persistValues();
+	}
+
+	private _barrierPosition: BarrierPosition | undefined;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Barrier Operator State",
+		states: enumValuesToMetadataStates(BarrierPosition)
+	})
+
+	public get barrierPosition(): BarrierPosition | undefined {
+		return this._barrierPosition;
+	}
+}
+
+@CCCommand(BarrierOperatorCommand.CapabilitiesGet)
+@expectedResponse(BarrierOperatorCCCapabilitiesReport)
+export class BarrierOperatorCCCapabilitiesGet extends BarrierOperatorCC {}
+
+@CCCommand(BarrierOperatorCommand.CapabilitiesReport)
+export class BarrierOperatorCCCapabilitiesReport extends BarrierOperatorCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+		
+		validatePayload(this.payload.length >=1);
+		const bitMaskLength = this.payload.length;
+		validatePayload(this.payload.length >= bitMaskLength);
+		this._supportedSignalTypes = parseBitMask(
+			this.payload.slice(0, bitMaskLength),
+			SignalType['Not Supported'],
+		);
```

Comment:
Not quite correct:
The payload length must be `>= 1 + bitMaskLength`, since the bitmask starts at the second byte.
The bitmask starts with value 1 at bit 0, so the 2nd argument to `parseBitMask` must be `SignalType.Audible`.

I'd probably remove the signal type `Not Supported` altogether if it is not used elsewhere.

Reply:
outdated lets see if the new update is what should be. Thank you for the detailed explanation